build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -0,0 +1,280 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fetch from "node-fetch";
+import { promises as fsPromises } from "fs";
+import { Flags } from "@oclif/core";
+import { Logger } from "@fluidframework/build-tools";
+import { BaseCommand } from "../../base";
+import { PackageVersionList } from "../../lib";
+import { isInternalVersionRange } from "@fluid-tools/version-tools";
+
+// Define the interface for build details
+interface IBuildDetails {
+	definition: { name: string };
+	status: string;
+	result: string;
+	sourceBranch: string;
+	finishTime: string;
+	buildNumber: string;
+}
+
+export class UnreleasedReportCommand extends BaseCommand<typeof UnreleasedReportCommand> {
+	static description = `Creates a release report for the most recent build published to an internal ADO feed. It does this by finding the most recent build in ADO produced from a provided branch, and creates a report using that version. The report always uses the "caret" report format.`;
+
+	static flags = {
+		repo: Flags.string({
+			description: "Repository name",
+			required: true,
+		}),
+		ado_pat: Flags.string({
+			description: "ADO Personal Access Token",
+			required: true,
+			env: "ADO_PAT",
+		}),
+		sourceBranch: Flags.string({
+			description: "Branch name across which the dev release manifest should be generated.",
+			required: true,
+		}),
+		output: Flags.string({
+			description: "Output manifest file path",
+			required: true,
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const flags = this.flags;
+
+		const repoName: string[] = flags.repo.split("/");
```

Comment:
Probably good to either check or assert that the length of the array is 2 and error out if not.